Scorep 6.0 in gompi 2020a doesn't work with binutils 2.34

The current Score-P-6.0-gompi-2020a.eb required for Scalasca doesn't work with the binutils 2.34 in gompi 2020a, the `bfd_get_section_flags` et.al. depreciation issue. 

Hear say has it, that this will be fixed in the next Score-P release.  As a bug fix we need to move the Score-P version in 2020a once the new version (or a beta) becomes publicly available.

Perhaps it is an idea to retire Score-P/Scalasca from the current distribution until the fix is available.

With out the patch we got error message like:

`libscorep_measurement.so: error: undefined reference to 'bfd_get_section_flags'`

at the link step when building instrumented executable with the instrumenter`scorep` or the depreciated scalasca instrumenter `skin`.

--- a/easybuild/easyconfigs/s/Score-P/Score-P-6.0-gompi-2020a.eb
+++ b/easybuild/easyconfigs/s/Score-P/Score-P-6.0-gompi-2020a.eb
@@ -1,6 +1,6 @@
 ##
 # This is an easyconfig file for EasyBuild, see https://github.com/easybuilders/easybuild
-# Copyright:: Copyright 2013-2019 Juelich Supercomputing Centre, Germany
+# Copyright:: Copyright 2013-2020 Juelich Supercomputing Centre, Germany
 # Authors::   Bernd Mohr <b.mohr@fz-juelich.de>
 #             Markus Geimer <m.geimer@fz-juelich.de>
 # License::   3-clause BSD
@@ -23,8 +23,10 @@ toolchain = {'name': 'gompi', 'version': '2020a'}
 
 source_urls = ['https://www.vi-hps.org/cms/upload/packages/scorep/']
 sources = ['scorep-%(version)s.tar.gz']
+patches = ['Score-P-6.0_binutils_2.34_api_change.patch']
 checksums = [
     '5dc1023eb766ba5407f0b5e0845ec786e0021f1da757da737db1fb71fc4236b8',  # scorep-6.0.tar.gz
+    'd5a5208077f60850d61a230c3a2d5e0d46ac0b5ec962904cba3010b9a904b59f',  # Score-P-6.0_binutils_2.34_api_change.patch
 ]
 
 dependencies = [

--- a/easybuild/easyconfigs/s/Score-P/Score-P-6.0_binutils_2.34_api_change.patch
+++ b/easybuild/easyconfigs/s/Score-P/Score-P-6.0_binutils_2.34_api_change.patch
@@ -1,0 +1,24 @@
+Support libbfd API change introduced with binutils 2.34.  Patch backported
+from upstream repo; fix will be included in releases after v6.0.
+--- a/src/services/unwinding/scorep_unwinding_unify.c
++++ b/src/services/unwinding/scorep_unwinding_unify.c
+@@ -28,6 +28,19 @@
+ 
+ #if HAVE( LIBBFD )
+ #include <bfd.h>
++
++#ifndef bfd_get_section_size
++#define bfd_get_section_size( asection ) bfd_section_size( asection )
++#endif
++
++#ifndef bfd_get_section_vma
++#define bfd_get_section_vma( abfd, asection ) bfd_section_vma( asection )
++#endif
++
++#ifndef bfd_get_section_flags
++#define bfd_get_section_flags( abfd, asection ) bfd_section_flags( asection )
++#endif
++
+ #endif
+ 
+ #include "scorep_unwinding_region.h"

